pkg/sql/ttl/ttljob/ttljob_test: TestInboundForeignKeyOnDeleteRestrictNull failed

pkg/sql/ttl/ttljob/ttljob_test.TestInboundForeignKeyOnDeleteRestrictNull [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/11956035?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/11956035?buildTab=artifacts#/) on master @ [3083a3057f2dcab041ba3fd7d463603351166728](https://github.com/cockroachdb/cockroach/commits/3083a3057f2dcab041ba3fd7d463603351166728):

Fatal error:

```
panic: test timed out after 4m55s
running tests:
	TestInboundForeignKeyOnDeleteRestrictNull (2s)

```
Stack: 

```
goroutine 239559 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2241 +0x31c
created by time.goFunc
	GOROOT/src/time/sleep.go:176 +0x38
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestInboundForeignKeyOnDeleteRestrictNull
    test_log_scope.go:167: test logs captured to: /artifacts/tmp/_tmp/772d54cb8543909a2bf6361ebbdd57e0/logTestInboundForeignKeyOnDeleteRestrictNull1734570714
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:82: cluster virtualization disabled due to issue: #109391 (expected label: C-bug)

pkg/sql/ttl/ttljob/ttljob_test_test/pkg/sql/ttl/ttljob/ttljob_test.go:129: (newRowLevelTTLTestJobTestHelper)
	NOTICE: .DB() called via implicit interface ApplicationLayerInterface;
HINT: consider using .ApplicationLayer().DB() instead.
TIP: consider replacing the test server initialization from:
    ts, ... := serverutils.StartServer(t, ...)
    defer ts.Stopper().Stop(...)
to:
    srv, ... := serverutils.StartServer(t, ...)
    defer srv.Stopper().Stop(...)
    ts := srv.ApplicationLayer()

See also: https://go.crdb.dev/p/testserver-and-cluster-virtualization
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestInboundForeignKeyOnDeleteRestrictNull.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-31873

--- a/pkg/sql/ttl/ttljob/BUILD.bazel
+++ b/pkg/sql/ttl/ttljob/BUILD.bazel
@@ -56,13 +56,14 @@ go_library(
 
 go_test(
     name = "ttljob_test",
+    size = "large",
     srcs = [
         "main_test.go",
         "ttljob_processor_test.go",
         "ttljob_query_builder_test.go",
         "ttljob_test.go",
     ],
-    args = ["-test.timeout=295s"],
+    args = ["-test.timeout=895s"],
     exec_properties = {"Pool": "medium"},
     deps = [
         ":ttljob",

